### PR TITLE
Map toString brackets

### LINKED ISSUE #8030: [std] use map literal syntax in Map's toString()
In the context of modern Haxe with its map literal syntax, `Map`'s string representation feels a bit unnatural, as it uses `{}` instead of `[]`:

```haxe
class Main {
	static function main() {
		trace([1 => "one"]); // {1 => one}
	}
}
```

I'm assuming that this is from a time before Haxe had map literals?

Changing it might be considered a minor breaking change.

### PR DESCRIPTION
**Pull request by RblSb**
Sorry for copy, i just recreated Haxe repo last week trying to trigger CI.
Closes #8030

### CODE REVIEW

File: tests/runci/targets/Cpp.hx
Comment (RealyUniqueName) [inline tests/runci/targets/Cpp.hx]: please, revert this.

**Review by RealyUniqueName — CHANGES_REQUESTED** [review]

**Comment by RealyUniqueName** [comment]
I'm not sure If we should treat this as a breaking change. I'd merge it for 4.1.

**Comment by RealyUniqueName** [comment]
CI disaster is not related.

@Simn are you ok with this for 4.2?

**Comment by Simn** [comment]
There are some failing tests on the C++ target here.

**Comment by RblSb** [comment]
This depends on https://github.com/HaxeFoundation/hxcpp/pull/892
Not sure if hxcpp changes are good tho

File: tests/runci/targets/Cpp.hx
Comment (kLabz) [reply tests/runci/targets/Cpp.hx]: Why did you "resolve" this?

Comment (kLabz) [reply tests/runci/targets/Cpp.hx]: Might have been automatic. Will need to have it reverted before merging in any case (but yeah it's fine to keep it around for tests on this PR)

**Comment by RblSb** [comment]
Jvm fail is not related. I can revert latest commit now and this probably good to merge (after https://github.com/HaxeFoundation/hxcpp/pull/892).
I'm not sure if all map cases are handled in hxcpp PR, with all map types and from dynamic cast cases, but this can be fixed with hxcpp releases if someone will find some `toString()` bug.

**Pull request merged**
